File: android/src/main/java/com/reactnativecommunity/viewpager/ViewPagerFragment.java
package com.reactnativecommunity.viewpager;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewParent;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import java.lang.ref.WeakReference;

import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;

public class ViewPagerFragment extends Fragment {
    private final int mPosition;
    private WeakReference<View> mReactViewRef;

    public ViewPagerFragment(int position, View child) {
        mPosition = position;
        mReactViewRef = new WeakReference<>(child);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        FrameLayout rootView = new FrameLayout(inflater.getContext());
        View reactView = mReactViewRef.get();
        if (reactView != null) {
            ViewParent previousParent = reactView.getParent();
            if (previousParent instanceof FrameLayout) {
                ((FrameLayout) previousParent).removeAllViews();
            }
            rootView.addView(reactView, MATCH_PARENT, MATCH_PARENT);
        }
        return rootView;
    }

    public int getPosition() {
        return mPosition;
    }

    public boolean onReactViewUpdate(FragmentAdapter adapter) {
        View reactView = adapter.getViewAtPosition(mPosition);
        if (reactView == null || reactView == mReactViewRef.get()) {
            return false;
        }
        mReactViewRef = new WeakReference<>(reactView);
        FrameLayout rootView = (FrameLayout) getView();
        if (rootView != null) {
            if (rootView.getChildCount() > 0) {
                rootView.removeAllViews();
            }
            rootView.addView(reactView, MATCH_PARENT, MATCH_PARENT);
        }
        return true;
    }
}

File: android/src/main/java/com/reactnativepagerview/ViewPagerFragment.kt
package com.reactnativepagerview

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.fragment.app.Fragment
import java.lang.ref.WeakReference

class ViewPagerFragment(private var mPosition: Int, child: View?) : Fragment() {
    private var mReactViewRef: WeakReference<View?>

    init {
        mReactViewRef = WeakReference(child)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val rootView = FrameLayout(inflater.context)
        val reactView = mReactViewRef.get()
        if (reactView != null) {
            detachFromParent(reactView)
            rootView.addView(reactView, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        }
        return rootView
    }

    fun getPosition(): Int {
        return mPosition
    }

    fun onReactViewUpdate(adapter: FragmentAdapter, positionDelta: Int): Boolean {
        mPosition += positionDelta
        val reactView = adapter.getViewAtPosition(mPosition)
        if (reactView == null || reactView === mReactViewRef.get()) {
            return false
        }
        mReactViewRef = WeakReference(reactView)
        val rootView = view as FrameLayout?
        if (rootView != null) {
            if (rootView.childCount > 0) {
                rootView.removeAllViews()
            }
            detachFromParent(reactView)
            rootView.addView(reactView, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        }
        return true
    }

    private fun detachFromParent(view: View) {
        val parent = view.parent
        if (parent is FrameLayout) {
            parent.removeView(view)
        }
    }
}
